.golangci.yml

```diff
@@ -1,343 +1,318 @@
-# - gochecknoinits    # Checks that no init functions are present in Go code.
-# - goerr113          # Go linter to check the errors handling expressions.
-# - gofumpt           # Gofumpt checks whether code was gofumpt-ed.
-# - inamedparam       # reports interfaces with unnamed method parameters.
-# - ireturn           # Accept Interfaces, Return Concrete Types.
-# - maintidx          # Maintidx measures the maintainability index of each function.
-# - musttag           # Enforce field tags in (un)marshaled structs.
-# - nestif            # Reports deeply nested if statements.
-# - nonamedreturns    # Reports all named returns.
-# - paralleltest      # Detects missing usage of t.Parallel() method in your Go test.
-#  - promlinter       # Check Prometheus metrics naming via promlint.
-# - rowserrcheck      # Checks whether Rows.Err of rows is checked successfully.
-# - sqlclosecheck     # Checks that sql.Rows and sql.Stmt are closed.
-# - wrapcheck         # Checks that errors returned from external packages are wrapped.
-# TODO: Need to discuss/decide/remove
-# - exhaustive     # Check exhaustiveness of enum switch statements.
-# - godox          # Tool for detection of FIXME, TODO and other comment keywords.
-# - inamedparam    # reports interfaces with unnamed method parameters.
-# - interfacebloat # A linter that checks the number of methods inside an interface.
-# - lll            # Reports long lines.
-# - stylecheck     # Stylecheck is a replacement for golint.
-# - tagliatelle    # Checks the struct tags.
-# - testpackage    # Linter that makes you use a separate _test package.
-
-issues:
-  exclude-dirs:
-    - pkg/api/v1alpha1/dynakube # legacy version, should not be changed
-  exclude-rules:
-    # Exclude duplicate code and function length and complexity checking in test
-    # files (due to common repeats and long functions in test code)
-    - path: _(test|gen)\.go
-      linters:
-        - dupl
-        - wsl
-        - perfsprint
-    - path: (test/*)
-      linters:
-        - noctx
-        - wsl
-        - thelper
-        - perfsprint
-        - govet
-    - path: _(test|gen)\.go
-      linters:
-        - govet
-        - dupl
-    - linters:
-        - gosec
-      path: pkg/webhook/validation/proxy_url_test.go
-    - linters:
-        - gosec
-      path: pkg/ingestendpoint/secret_test.go
-    - linters:
-        - unparam
-      text: always receives
-    - linters:
-        - revive
-      path: pkg/clients/dynatrace # it's awaiting refactoring
-    # Run some linter only for test files by excluding its issues for everything else.
-    - path-except: 'pkg/api/(.+)\.go'
-      linters:
-        - godot
+    - gci
+    - gofmt
+  exclusions:
+    generated: lax
+    paths:
+      - pkg/api/v1alpha1/dynakube
+      - pkg/api/v1alpha1/edgeconnect
+      - pkg/api/v1beta1/dynakube
+      - pkg/api/v1beta2/dynakube
+      - pkg/api/v1beta3/dynakube
+      - third_party$
+      - builtin$
+      - examples$
```

Comment:
what is examples?

Reply:
it was migrated, dunno

I only added the `/pkg/api..` stuff (and the `staticcheck` parts), everything else was migrated via `golangci-lint migrate`